code/controllers/subsystem/movement/zfall.dm

```diff
@@ -0,0 +1,37 @@
+/// Responsible for triggering mob zfalls when they are not already triggered by structure deconstruction or moving into openspace
+SUBSYSTEM_DEF(zfall)
+	name = "Zfall"
+	flags = SS_TICKER
+	wait = 5
+	runlevels = RUNLEVEL_GAME|RUNLEVEL_POSTGAME
+	priority = FIRE_PRIORITY_ZFALL
+	var/list/atom/movable/openspace_inhabitants = list()
+
+/datum/controller/subsystem/zfall/proc/add_openspace_inhabitant(atom/movable/A)
+	if(!ismob(A) || (A in openspace_inhabitants))
+		return
+	RegisterSignal(A, COMSIG_PARENT_QDELETING, .proc/remove_openspace_inhabitant, A)
+	openspace_inhabitants |= A
+
+/datum/controller/subsystem/zfall/proc/remove_openspace_inhabitant(atom/movable/A)
+	SIGNAL_HANDLER
+	if(!(A in openspace_inhabitants))
+		return
+	UnregisterSignal(A, COMSIG_PARENT_QDELETING)
+	openspace_inhabitants -= A
+
+/datum/controller/subsystem/zfall/fire(resumed)
+	var/list/removals = list()
+	for(var/atom/movable/A as() in openspace_inhabitants)
+		var/turf/T = get_turf(A)
+		if(!isopenspace(T))
+			removals |= A
```

Comment:
You might as well remove it immediately if you are going to copy the list in `for(var/atom/movable/A as() in openspace_inhabitants)` anyway. The only reason you would do it this way is if you are using index based iteration (Which would be faster).
Also using `|=` unnecessarilly will be worse for performance.

Reply:
I wasn't aware that DM copied the list on iteration - in most languages I've used foreach loops don't allow concurrent modification

Reply:
Climbing removed entirely, however the code will remain

Reply:
Fixed this